### ISSUE TITLE
Allow use of zsh completions or integrate into Warp completions.

### ISSUE THREAD

**Issue by dannyneira** [warpdotdev/Warp#1953]
### Describe the solution you'd like?

```text
Lots of commands do not autocomplete because Warp by pass zsh completions. Please allow the use of zsh completions or integrate those completions in your autocomplete engine.
```

### Is your feature request related to a problem? Please describe.

```text
I've been very vocal on several platforms (Twitter, GitHub, discord) about the fact that rebuilding something that already exists, everybody uses, and is that personal like completions is a waste of development resources. Warp will never get it right and the 80/20 rule doesn't apply here. I, for example, use several tools a lot for which you certainly are not going to build completions, so by using Warp I will always be left without. Also, you will never get it right. People will always have different versions of the tools with different parameters installed. Will you build completions for all those tools and all those versions? How are you going to keep this up to date? You are wasting your resources on something that already exists, is working and accepted, and is in use.
Building yet another "standard" for building your own completions doesn't make sense either. People need to learn yet another thing, just to be able to use it only with Warp. And again, completions already exist, there is already a standard. Spend your resources on other features.
```

### How important is this feature to you?

5 (Can't work without it!)

**Comment by dannyneira** [comment]
Please +1 this request as it helps us gauge interest. We will post any updates here.

**Comment by varenc** [comment]
I'm a little confused because https://github.com/warpdotdev/Warp/issues/150 was closed by alokedesai with the comment that "Marking this closed as we should support completions for functions in the latest release".  

But this issue says otherwise and I can confirm from my own experience that custom completions still aren't supported.  Both completions for my own private functions and custom completions for system utilities that don't come with their own completions.  `networksetup` for example doesn't come with completions but I hacked one together myself.

Let's focus on the `networksetup` example. My completion for it works great in a normal terminal but no luck in Warp. Here's the completion definition:

```bash
$ which _networksetup
_networksetup () {
	local cur prev opts opts_ns
	COMPREPLY=()
	cur="${COMP_WORDS[COMP_CWORD]}"
	prev="${COMP_WORDS[COMP_CWORD-1]}"
	opts="-listnetworkserviceorder -listallnetworkservices -listallhardwareports         ..."
        ....
        ....
        if [[ $(grep -c "${prev:1}" <<<"$dev_opts" ) -eq 1 ]]
	then
		dev_names=$(networksetup -listallhardwareports  | grep "Device" | cut -f 2 -d ' ')
		COMPREPLY=($(compgen -W "$dev_names" - "$cur"))
		return
	fi
	...
```
Full completion definition here:  https://app.warp.dev/block/eu7NWoZzT5HoHOcKKn9HeK  (also note I'm very sure my completion isn't great and shows a lack of depth of understanding... but it works for me and it's hugely useful for remembering all the wordy options it takes)

So anyway, here's another vote for Warp supporting custom user-defined completions as defined in zsh's own completion system.  I understand Warp's completions, when created, are far superior, but I don't expect to Warp to ever create their own completions for obscure things like `networksetup` or for my own functions.
